### PR TITLE
fates api 24

### PR DESCRIPTION
**Pull request by rgknox**
This set of changes enables coupling with FATES API 24.  This introduces the ability of FATES to dictate the number of veg-patches used on the natural land unit, an updated FATES parameter file format, hooks to enable fates wood harvest product fluxes, and hooks for a new crown damage module.

### CODE REVIEW

**Comment by rgknox** [comment]
land developer tests on cori pass:
/global/cscratch1/sd/rgknox/e3sm_tests/E2ba9fa8333-Fdef6b3e7-fates-api24-v2

All fates tests have expected NLCOMP and BASELINE differences
All non-fates tests are B4B with base with some timing differences which I believe are due to machine performance variability

File: cime_config/tests.py [hunk 1]
```diff
@@ -57,6 +57,7 @@
             "SMS_Ld20.f45_f45.IELMFATES.elm-fates_rd",
             "SMS_Ld20.f45_f45.IELMFATES.elm-fates_eca",
             "SMS_Ld30.f45_f45.IELMFATES.elm-fates_satphen",
+            #"SMS_Ld30.f45_f45.IELMFATES.elm-fates_damage", # tree-damage coming soon
```
Comment (bishtgautam) [inline cime_config/tests.py:60]: Can this test be turned on?

[the remark this answers is not in the document]
Comment (rgknox) [reply cime_config/tests.py:60]: not yet, the model will gracefully fail.  The fates tree damage module will be in soon though.  I can remove this if its clutter, no problem.

**Comment by rgknox** [comment]
Thanks for comments @bishtgautam , I think I've addressed them

**Review by bishtgautam — APPROVED** [review]

**Comment by bishtgautam** [comment]
@rgknox Is this PR BFB?

**Review by bishtgautam — CHANGES_REQUESTED** [review]
Since the new test is not active, can you please remove `components/elm/cime_config/testdefs/testmods_dirs/elm/fates_damage`?

**Comment by rgknox** [comment]
I can. This module is in progress in FATES, we do plan on having a test for it.

**Comment by rgknox** [comment]
@bishtgautam , removed the damage test defs

**Review by bishtgautam — APPROVED** [review]

**Comment by rljacob** [comment]
@bishtgautam is this ready to merge?

**Comment by peterdschwartz** [comment]
I've merged to my local next branch and am just waiting on test suite results.

**Review by bishtgautam — APPROVED** [review]

**Comment by peterdschwartz** [comment]
merged to next

**Comment by peterdschwartz** [comment]
Merged to master.

**Pull request merged**